### ISSUE TITLE
Compat with season mods? 

### ISSUE THREAD

**Issue by JorisDeBeer** [lukebemishprojects/Tempest#20]
Hello, maybe it would be a good idea to make build in compatibility with season and temperature mods? Would make it really immersive.

**Comment by lukebemish** [comment]
Fabric seasons does not expose any API I can use. Serene seasons is distributed under a restrictive ARR license and thus hard to add compat for without risking issues in the future. As for temperature mods - that really has to be done on their end given what's involved in most cases. I expose an API people can use. If there's specific temperature mods that expose an API I could work with from my end I can look into it but generally speaking it's a lot easier to implement from the other end.

**Comment by Steveplays28** [comment]
Hi Luke, I'm working on a new seasons mod with Joris, titled [Simple Seasons](https://github.com/Steveplays28/simple-seasons). Would you be up to work together to add compatibility?

I think it'd be super cool to have seasons and regional weather working together, It's something I've wanted to make happen for quite a while!

**Comment by lukebemish** [comment]
I can add compatibility for a seasons mod pretty easily on my end - basically, I just need some API I can query to get the current season from your mod, and then I'll use that when calculating the weather.

**Comment by Steveplays28** [comment]
Awesome, thanks. I'll get that in place for you.
I'm also going to look into adding NeoForge multiloader support for 1.20.2 and up for Simple Seasons.

I'll let you know how it goes!

**Comment by lostopkk** [comment]
> Hi Luke, I'm working on a new seasons mod with Joris, titled [Simple Seasons](https://github.com/Steveplays28/simple-seasons). Would you be up to work together to add compatibility?
> 
> I think it'd be super cool to have seasons and regional weather working together, It's something I've wanted to make happen for quite a while!

simple seasons looks very promising! I'm looking forward to the stable version of this mod :)

**Comment by Steveplays28** [comment]
@lostopkk Thanks, glad you like it!

> I can add compatibility for a seasons mod pretty easily on my end - basically, I just need some API I can query to get the current season from your mod, and then I'll use that when calculating the weather.

@lukebemish I added an [API for fetching the current season per world](https://github.com/Steveplays28/simple-seasons/commit/5b558f52cc2bbd25114ff4c233d649fab0c58caa). It simply returns the season int, which you can check against the `SimpleSeasons#Seasons` enum, using the value of the enum members.
I still have to make the season be per world instead of serverwide, but I did add the `World` as an argument for the API so you don't have to worry about changing your implementation immediately when I get to that. I also still need to add JavaDocs.
Does that look good?

Also, FYI, the [`1.20.x` branch](https://github.com/Steveplays28/simple-seasons/tree/1.20.x) is currently where the main development is happening. I'll switch the branches around sometime. You'll also have to use local Gradle dependencies, see this example: [1](https://github.com/Steveplays28/blendium/tree/ba7738b08deb0673673054703720faabe3f89794/libraries), [2](https://github.com/Steveplays28/blendium/blob/ba7738b08deb0673673054703720faabe3f89794/build.gradle#L29C1-L30C32), and [3](https://github.com/Steveplays28/blendium/blob/ba7738b08deb0673673054703720faabe3f89794/build.gradle#L53-L54).

For temperature and such, I'm planning on modifying the actual biomes for that, but I still have to look into exactly how to do that. So far I'm only changing precipitation and no actual simulation of temperatures is in place yet. Might be cool to look into for the future, but that might get tricky with compat with Tempest.
If you have any ideas on that, let me know.

**Comment by SaltSouls** [comment]
Would be nice if we could also get serene seasons support. I believe their mod already has a seasons api that you can hook into.

**Comment by lukebemish** [comment]
Serene seasons is distributed under a restrictive ARR license and thus hard to add compat for without risking potential legal issues in the future.

**Comment by SaltSouls** [comment]
> Serene seasons is distributed under a restrictive ARR license and thus hard to add compat for without risking potential legal issues in the future.

Really? [Cold Sweat](https://modrinth.com/mod/cold-sweat) has built-in compatibility for it and I haven't heard/seen anything about that being an issue. They even have a whole API section in their code, I assume for people to add interoperability between their mods and Serene Seasons. I believe the legal issues would only occur if you started to try and rip their code/assets and use them in your own mod, instead of using their APIs.

**Comment by lukebemish** [comment]
They (Serene Seasons) distribute their mod under an "all rights reserved" license. Any mods that depend on their API do so entirely at their own risk and are at constant risk of being told they are violating Serene Season's license

**Comment by mosharky** [comment]
Their API is supposedly public according to Glitchfiend/SereneSeasons#448. Will compatibility be considered here then?

**Comment by lukebemish** [comment]
As I already said, 

> They (Serene Seasons) distribute their mod under an "all rights reserved" license. Any mods that depend on their API do so entirely at their own risk and are at constant risk of being told they are violating Serene Season's license

Whether the API is public or not is unimportant. It is distributed under an ARR license.

**Comment by mosharky** [comment]
Fair enough I guess, but if it's something you'd like, they're also open to giving explicit permission:
![image](https://github.com/lukebemishprojects/Tempest/assets/32606683/6c2a5b36-3bf4-48ec-b09b-ea57993fcf68)

Not sure how much that holds up against an ARR license though

**Comment by mosharky** [comment]
Any updates on this?

**Comment by cintreck** [comment]
would be great to get that serene seasons support

**Comment by lukebemish** [comment]
It's WIP; I've been quite busy with real-life stuff and other projects, so work on Tempest has been delayed, but I'll hopefully get it updated, and add some compat, in the next month or so.

**Comment by cintreck** [comment]
oh great, thank you!

**Comment by autumnly24** [comment]
Support for the Ecliptic Seasons mod would be great, they have a pretty accessible API, a lot of documentation, and it's open source and maintained for a ton of versions of the game!
